activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -3087,6 +3087,24 @@ def test_has_many_association_with_same_foreign_key_name
     end
   end
 
+  def test_key_ensuring_owner_was_is_not_valid_without_dependent_option
+    error = assert_raises(ArgumentError) do
+      Class.new(ActiveRecord::Base) do
+        has_many :books, ensuring_owner_was: true
+      end
+    end
+
+    assert_match(/Unknown key: :ensuring_owner_was/, error.message)
+  end
+
+  def test_key_ensuring_owner_was_is_valid_when_dependent_option_is_destroy_async
+    Class.new(ActiveRecord::Base) do
+      self.destroy_association_async_job = Class.new
+
+      has_many :books, ensuring_owner_was: true, dependent: :destroy_async
```

Comment:
Let's flip the argument order, since that's more akin to what you'd do in an app — and then apply the `:destroyed?` thing from the other comment.

```suggestion
      has_many :books, dependent: :destroy_async, ensuring_owner_was: :destroyed?
```